### ISSUE TITLE
Should bluez rpi version be a dependecy anymore?

### ISSUE THREAD

**Issue by mibofra** [RPi-Distro/pi-bluetooth#15]
As learnt from here: https://github.com/RPi-Distro/pi-bluetooth/issues/14 , bdaddr is in bluez rpi version of the package. 
But with the new master [commit](https://github.com/RPi-Distro/pi-bluetooth/commit/03396693cf1001cf56d9834e7ed21d5ef04739ae) not using it anymore, should the rpi version of bluez be maintained as a dependecy in the debian package, or bluez rpi version provides something more?

**Comment by XECDesign** [comment]
It could probably be downgraded to the +rpt1 version, but there are other changes not in the Debian package.

**Comment by mibofra** [comment]
I've tested the latest versions of the bt support scripts with bluez from vanilla debian @XECDesign , the bluetooth is working fine. What is in bluez modded version to still be a dependency? (I could just to a diff between the two packages)

**Comment by XECDesign** [comment]
[Add-avdtp_set_priority.txt](https://github.com/RPi-Distro/pi-bluetooth/files/5090762/Add-avdtp_set_priority.txt)
[raspberry-pi-mods.txt](https://github.com/RPi-Distro/pi-bluetooth/files/5090763/raspberry-pi-mods.txt)

**Comment by mibofra** [comment]
@XECDesign https://github.com/RPi-Distro/firmware-nonfree/issues/8#issuecomment-673309390 this comment, the user used bluez_5.50-1.2~deb10u1_armhf.deb from debian. 
However, it's pretty specific, the suggestion is to make another bluez package, like bluez-rpi, not replacing the one from vanilla debian, but making it conflicting and providing it. So if bluez-rpi is installed, bluez is not installed. 

When building the rpi image, having simply bluez-rpi in the packages list, will install it instead of bluez.

Example usage: other distros are not rpios, can import them directly of using rpios repos, to use them, without having package conflicts. Maybe that isn't the usual case, but it can be useful too.

**Comment by seamusdemora** [comment]
I have **never** had any luck getting BT on RPi3B+ to do anything useful. I gave up trying abt 2 years ago, after applying numerous *"snake-oil"* fixes. I've just seen a [post on RPi SE](https://raspberrypi.stackexchange.com/questions/116651/ble-connection-timeout-on-pi3-b) wherein the OP implied he fixed his issues by applying [this recipe](https://github.com/RPi-Distro/firmware-nonfree/issues/8#issuecomment-673309390). It is a mystery to me how the OP "fixed" his problem (also with a RPi3B+) with this recipe, but since nothing else has worked, I tried it also :P 

Here's a synopsis: 
```bash
pi@raspberrypi3b:~ $ apt info firmware-brcm80211 | grep Version
Version: 1:20190114-1+rpt7 
pi@raspberrypi3b:~ $ dpkg --status bluez | grep '^Version:'
Version: 5.50-1.2~deb10u1+rpt2
pi@raspberrypi3b:~ $ wget http://ftp.hk.debian.org/debian/pool/main/b/bluez/bluez_5.50-1.2~deb10u1_armhf.deb
...
Saving to: ‘bluez_5.50-1.2~deb10u1_armhf.deb’
pi@raspberrypi3b:~ $ sudo apt install ./bluez_5.50-1.2~deb10u1_armhf.deb
...
Note, selecting 'bluez' instead of './bluez_5.50-1.2~deb10u1_armhf.deb'
...
The following packages will be REMOVED:
  pi-bluetooth
The following packages will be DOWNGRADED:
  bluez
...
Do you want to continue? [Y/n] Y
...
Removing pi-bluetooth (0.1.14) ...
dpkg: warning: downgrading bluez from 5.50-1.2~deb10u1+rpt2 to 5.50-1.2~deb10u1
...
pi@raspberrypi3b:~ $ dpkg --status bluez | grep '^Version:'
Version: 5.50-1.2~deb10u1
```

In summary, it seems I've *downgraded* **FROM: ** `Version: 5.50-1.2~deb10u1+rpt2` **TO: ** `Version: 5.50-1.2~deb10u1`. 

Since making these changes, I've not tried to make any BT connections - my confidence is low that this will improve things. Can anyone offer any enlightenment, or is BT on Pi3B+ simply a hopeless cause?

**Comment by sebastianeferrer** [comment]
> Since making these changes, I've not tried to make any BT connections - my confidence is low that this will improve things. Can anyone offer any enlightenment, or is BT on Pi3B+ simply a hopeless cause?

Dear @seamusdemora be happy now! This bugfix will through light to you solving the conflicts between wifi and bt on any Rpi model using CYW43455.

Check it out here -> https://github.com/RPi-Distro/firmware-nonfree/issues/8

**Comment by seamusdemora** [comment]
@sebastianeferrer 👍  Thank you for the update - I've added a comment to the [referenced RPi SE Q&A](https://raspberrypi.stackexchange.com/questions/116651/ble-connection-timeout-on-pi3-b). 

I will take your advice and "_be happy now_" :)   However... issues with Bluetooth on RPi go well beyond the WiFi-BT interference issue. To _blether_ just a wee bit: 

1. I finally did make some progress wrt BT on my RPi system [REF this Q&A](https://raspberrypi.stackexchange.com/questions/116819/bluetooth-blues-redux). I can state _unequivocally_ that it had nothing to do with WiFi (it's not enabled on this RPi). I feel the issue is rather one of poor documentation, and (perhaps) some dodgy software! Why should BT be virtually unusable on a headless RPi (_RPi OS Lite_) when it is largely a "point-and-click" operation in the GUI?? 
2. I don't know any of the details on the changes you've made in the bugfix. If it helps those who use BT and WiFi together, then _bully for you_! :)   But as I understand it: **_all RPis share their 2.4 GHz antenna with WiFi and BT_**, and if that's the case, there will always be issues that software cannot overcome.
